### PR TITLE
Find command test

### PR DESCRIPTION
**Pull request by ChenXy128**
#63

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+     */
+    private PersonFieldsContainKeywordPredicate preparePredicate(String... categoryDescriptionPairs) {
+        HashMap<String, String> categoryDescriptionMap = new HashMap<>();
+        for (int i = 0; i < categoryDescriptionPairs.length; i += 2) {
+            categoryDescriptionMap.put(categoryDescriptionPairs[i], categoryDescriptionPairs[i + 1]);
+        }
+        return new PersonFieldsContainKeywordPredicate(categoryDescriptionMap, new HashSet<>());
+    }
+}
+
```
Comment (NusMinato) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:148]: Good job! Nice implementation of Find command test. I can refer to your code hahaha.

**Review by NusMinato — COMMENTED** [review]
Looks good to me. Nice implementation of Find command test.

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    /**
+     * Tests the equality of the FindCommand. Verifies that:
+     * - A FindCommand equals itself.
+     * - Two FindCommand objects with the same predicate are equal.
+     * - A FindCommand is not equal to null.
+     * - A FindCommand is not equal to an object of a different type.
+     * - Two FindCommand objects with different predicates are not equal.
+     */
+    @Test
+    public void equals() {
```
Comment (HongRay) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:50]: Nice equals test method written. Will use it for edit command.

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+public class FindCommandTest {
+    private Model model;
+    private Model expectedModel;
+    /**
+     * Sets up the model and expectedModel with a typical address book before each test.
+     * This ensures a consistent starting state for tests.
+     */
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
```
Comment (HongRay) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:38]: Our `getTypicalAddressBook()` is not done.

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    public void execute_nonExistingCategoryAndDescription_noPersonFound() {
+        String expectedMessage = String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, 0);
+        PersonFieldsContainKeywordPredicate predicate = preparePredicate("NilCategory", "NilDescription");
+        FindCommand command = new FindCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertTrue(model.getFilteredPersonList().isEmpty());
+    }
+    /**
+     * Helper method to prepare a {@link PersonFieldsContainKeywordPredicate} based on the provided category
```
Comment (Xuyan0518) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:133]: LGTM. Nice job on adding in all the assertions.

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+     * and description pairs. Facilitates easier creation of predicates for testing.
+     *
+     * @param categoryDescriptionPairs Varargs input where each pair of strings represent
+     *     a category and its corresponding description.
+     * @return A {@code PersonFieldsContainKeywordPredicate} created from the specified category and description pairs.
+     */
+    private PersonFieldsContainKeywordPredicate preparePredicate(String... categoryDescriptionPairs) {
```
Comment (Xuyan0518) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:140]: Good job on writing all the necessary tests. They all look good to me.

**Pull request closed without merging**